src/ArborX_LinearBVH.hpp

```diff
@@ -178,6 +181,8 @@ BasicBoundingVolumeHierarchy<MemorySpace, Value, IndexableGetter,
                                               ExecutionSpace>::value,
                 "Primitives must be accessible from the execution space");
   constexpr int DIM = GeometryTraits::dimension_v<BoundingVolume>;
+  using FloatingPoint =
+      typename GeometryTraits::coordinate_type<BoundingVolume>::type;
```

Comment:
I would prefer if you deduced these closer to the place you actually use them.  As far as I can tell there is really one place.